### PR TITLE
docs: add descriptions to the example codes in "FEATURES"

### PR DESCRIPTION
**Pull request by graczhual**
Split a whole block of example code and add explanations for different input
parameters to improve the readability.

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2gh5pf4 所有样例代码增加不同输入参数的打印结果和描述](https://app.clickup.com/t/2gh5pf4)

File: docs/source/features/dataset_management.rst [hunk 1]
```diff
@@ -19,12 +19,18 @@ operations require a dataset first. Of course, it is necessary to initialize a
...
 
 SDK provides method :meth:`~graviti.manager.dataset.DatasetManager.create` to support creating
-a dataset. In addition to the name, alias and storage config can also be specified:
+a dataset based on the given name:
 
 .. code:: python
 
    ws.datasets.create(f"{DATASET_NAME}")
 
+In addition to the name, alias and storage config can also be specified:
```
Comment (Lee-000) [inline docs/source/features/dataset_management.rst:28]: `In addition to name`?

File: docs/source/features/version_control/branch.rst [hunk 2]
```diff
@@ -96,7 +100,8 @@ version of the dataset by branches. The version of dataset can be viewed by ``da
    dataset.HEAD
 
 .. note::
-   This ``checkout`` method is often used to update the version of the local dataset, for example:
+   This ``checkout branch`` method is often used to update the version of the local dataset,
```
Comment (Lee-000) [inline docs/source/features/version_control/branch.rst:103]: Is there a `checkout branch` method? It's just a `checkout` method.

File: docs/source/features/version_control/draft.rst [hunk 3]
```diff
@@ -23,18 +23,24 @@ in the SDK. Of course, it is necessary to get a dataset first:
...
 
-   dataset.drafts.create(f"{DRAFT_TITLE}")
-   dataset.drafts.create(f"{DRAFT_TITLE}", f"{DRAFT_DESCRIPTION}")
    dataset.drafts.create(f"{DRAFT_TITLE}", branch=f"{BRANCH_NAME}")
 
-.. note::
-   If no branch is specified, the created draft will be based on the current branch of the
-   dataset, which can be viewed by ``dataset.HEAD``.
+If no branch is specified, the created draft will be based on the current branch of the
+dataset, which can be viewed by ``dataset.HEAD``:
```
Comment (Lee-000) [inline docs/source/features/version_control/draft.rst:33]: Do we need to add a link to the API reference of `HEAD`?

File: docs/source/features/version_control/draft.rst [hunk 4]
```diff
@@ -23,18 +23,24 @@ in the SDK. Of course, it is necessary to get a dataset first:
...
+
+.. code:: python
+
+   dataset.drafts.create(f"{DRAFT_TITLE}")
+
+In addition to the title, it is also allowed to add description to the draft:
```
Comment (Lee-000) [inline docs/source/features/version_control/draft.rst:39]: Do we need `the` before `title`?

File: docs/source/features/version_control/draft.rst [hunk 5]
```diff
@@ -101,15 +115,20 @@ of the dataset.
 
 SDK provides method :meth:`~graviti.manager.draft.Draft.commit` to support committing a draft.
 This action means that a new commit will be created and all the changes from the draft will be
-saved into this commit.
+saved into this commit. Furthermore, it is not allowed to read or upload data on a closed draft:
```
Comment (Lee-000) [inline docs/source/features/version_control/draft.rst:118]: `on a committed draft`?

File: docs/source/features/version_control/tag.rst [hunk 6]
```diff
@@ -22,26 +22,31 @@ Before operating tags, a dataset with existing commits is needed:
...
-   If no tag is specified, the created tag will be based on the current commit of the
-   dataset, which can be viewed by ``dataset.HEAD.commit_id``.
+The revision can also be the branch name. In this situation, the tag will be created based
+on the latest commit of the branch:
+
+.. code:: python
+
+   dataset.tags.create(f"{TAG_NAME}", f"{BRANCH_NAME}")
+
+The revision can also be the tag name. SDK supports create multiple tags based one the same commit:
```
Comment (Lee-000) [inline docs/source/features/version_control/tag.rst:38]: `based on`?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

File: docs/source/features/version_control/draft.rst (hunk 3 above)
Comment (graczhual) [reply docs/source/features/version_control/draft.rst:33]: The description about HEAD will be modified together in another pr

File: docs/source/features/version_control/commit.rst [hunk 7]
```diff
@@ -23,34 +23,58 @@ First of all, it is necessary to get a dataset:
...
+
+Or the tag name:
+
+.. code:: python
+
    dataset.commits.list(f"{TAG_NAME}")
 
-.. note::
-   If no revision is specified, all commits preceding the current commit wiil be returned.
+If no revision is specified, all commits preceding the current commit wiil be returned:
```
Comment (Lee-000) [inline docs/source/features/version_control/commit.rst:44]: `will`

File: docs/source/features/version_control/commit.rst [hunk 8]
```diff
@@ -23,34 +23,58 @@ First of all, it is necessary to get a dataset:
...
+
+Or the tag name:
+
+.. code:: python
+
    dataset.commits.get(f"{TAG_NAME}")
 
-.. note::
-   If no revision is specified, the current commit of dataset wiil be returned.
+If no revision is specified, the current commit of dataset wiil be returned:
```
Comment (Lee-000) [inline docs/source/features/version_control/commit.rst:73]: will

File: docs/source/features/version_control/tag.rst [hunk 9]
```diff
@@ -22,26 +22,31 @@ Before operating tags, a dataset with existing commits is needed:
...
-   If no tag is specified, the created tag will be based on the current commit of the
-   dataset, which can be viewed by ``dataset.HEAD.commit_id``.
+The revision can also be the branch name. In this situation, the tag will be created based
+on the latest commit of the branch:
+
+.. code:: python
+
+   dataset.tags.create(f"{TAG_NAME}", f"{BRANCH_NAME}")
+
+The revision can also be the tag name. SDK supports create multiple tags based on the same commit:
```
Comment (Lee-000) [inline docs/source/features/version_control/tag.rst:38]: `supports creating`?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**